### DISCUSSION TITLE
How is the status about True PWM support?

### DISCUSSION THREAD

**Discussion by Ncerzzk** [ExpressLRS/ExpressLRS#1782] [General]
Hi, I notice that for now, the `True PWM` mode is not supported yet.

I'm trying to DIY a ELRS-pwm RX with true pwm with esp32-pico d4 ( to driver brush motor by a mos directly, the soft pwm rate is too low for this case). So if 
the true pwm support is on the way, I would not make the wheels again, ^-^ .

If not, then maybe I can try to handle it then make a pull request? Thank you.

**Comment by pkendall64** [comment]
There is already PWM support, both Matek and Betafpv sell PWM ELRS receivers.
Also, in the upcoming 3.0 release we have even better support for PWM with 8/12 and 16 full-res channels.

**Reply by Ncerzzk** [reply]
Thank you. 
I looked source code just now, for esp32, the pwm output is using LEDC which could satisfied my requiment.

while the wiki said that: 

> PWM frequency is currently locked at 50Hz.

can we adjust the pwm frequency now?

**Reply by pkendall64** [reply]
As part of 3.0 the update frequency is adjustable for each PWM output individually.

**Reply by Ncerzzk** [reply]
Fine, that's great ^ ^
